test/Properties/NumericProp.hs

```diff
@@ -0,0 +1,29 @@
+{-# LANGUAGE ImpredicativeTypes #-}
+
+module Properties.NumericProp (tests) where
+
+import Plutarch.Extra.Numeric ((#^))
+import Test.QuickCheck (
+  NonNegative (NonNegative),
+  Property,
+  (===),
+ )
+import Test.Tasty (TestTree, adjustOption, testGroup)
+import Test.Tasty.QuickCheck (QuickCheckTests, testProperty)
+
+tests :: TestTree
+tests =
+  adjustOption go $
+    testGroup
+      "Numeric"
+      [ testProperty "Integer power (#^)" propPowInt
+      ]
+  where
+    go :: QuickCheckTests -> QuickCheckTests
+    go = max 1000
+
+-- Properties
+
+propPowInt :: NonNegative Integer -> NonNegative Integer -> Property
+propPowInt (NonNegative n) (NonNegative i) =
+  (n ^ i) === plift (pconstant n #^ pconstant i)
```

Comment:
If we want to test exponentiation, that has laws of its own:

* `x ^ i * x ^ j = x ^ (i + j)`
* `(x ^ y) ^ z = x ^ (y * z)`
* `x ^ 1 = x`
* `x ^ 0 = 1`

These are actually demonstrating a semiring-like structure on exponentiation, hence the two identities (additive and multiplicative), and showing that we have a 'faithful' translation.